core/opds_import.py

```diff
@@ -909,25 +914,41 @@ def import_edition_from_metadata(self, metadata):
 
         return edition
 
-    def update_work_for_edition(self, edition):
+    def update_work_for_edition(
+        self,
+        edition: Edition,
+        is_open_access=True,
+    ) -> tuple[LicensePool | None, Work | None]:
```

Comment:
Nitpicking, but this is more of `tuple[Licensepool, Work] | tuple[None, None]`

Reply:
It is possible for `pool.calculate_work()` on l. 958 to return `None` for the work, so it at least seems possible to return `tuple[LicensePool, None]`, making the proposed hint overly restrictive.

I could tighten up the current hint as `tuple[Licensepool, Work] | tuple[LicensePool, None] | tuple[None, None]`, if you think that makes sense.

Reply:
That's overly complicated I think. We can just leave it as is